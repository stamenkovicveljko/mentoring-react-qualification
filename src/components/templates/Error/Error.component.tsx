import React, { ReactElement } from "react";
import * as Styled from "./Error.styles";

interface Props {
  title?: string;
  subtitle?: string;
  children?: React.ReactNode;
}

export default function ErrorTemplate({
  title = "Ooops, seems there is an error!",
  subtitle = "404 - Not Found",
  children,
}: Props): ReactElement {
  return (
    <Styled.Container>
      <h1>{title}</h1>
      <h2>{subtitle}</h2>
      {children}
    </Styled.Container>
  );
}
