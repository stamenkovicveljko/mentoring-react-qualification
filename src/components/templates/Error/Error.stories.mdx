import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import Error from "./Error.component";
import Link from "../../atoms/Link/Link.component";

<Meta title="Error" component={Error} />

# Error Component

Basic error component that display `404 - Not Found` by default.

## Implementation

Simply change `title`, `subtitle` or supply children to get customized error.

### Default

<Preview>
  <Story name="default">
    <Error />
  </Story>
</Preview>

### Custom error

<Preview>
  <Story name="401 - unauthorized">
    <Error title="There was an error." subtitle="401 - Unauthorized">
      Click on this <Link to="https://google.com">link</Link> to login.
    </Error>
  </Story>
</Preview>

## Props

<Props of={Error} />
