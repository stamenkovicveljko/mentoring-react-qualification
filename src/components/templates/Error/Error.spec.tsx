import React from "react";
import renderer from "react-test-renderer";
import Error from "./Error.component";

test("Error renders as expected", () => {
  const component = renderer.create(
    <Error title="There was an error." subtitle="401 - Unauthorized">
      Try logging in.
    </Error>
  );

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
