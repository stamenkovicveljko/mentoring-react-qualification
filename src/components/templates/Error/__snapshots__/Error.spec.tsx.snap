// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders as expected 1`] = `
<div
  className="sc-AxjAm ctoqlO"
>
  <h1>
    There was an error.
  </h1>
  <h2>
    401 - Unauthorized
  </h2>
  Try logging in.
</div>
`;
